src/net.cpp

```diff
@@ -2785,16 +2785,16 @@ void CConnman::ThreadOpenAddedConnections()
         bool tried = false;
         for (const AddedNodeInfo& info : vInfo) {
             if (!info.fConnected) {
-                if (!grant.TryAcquire()) {
+                if (!grant) {
```

Comment:
2e0f8a6d39a15ecac60b9d463606e16ba5707bb5: I'm confused by this change. I see `!fgrant` calls the `bool()` operator which checks `fHaveGrant`. But if that returns false we now break out of the `for` loop. Wheveras we used to call `TryAcquire()` and continue if that worked.

Reply:
> before entering the loop

Oh I see. The `CSemaphoreGrant grant(*semAddnode)` constructor before the loop doesn't `fTry`, but uses `Acquire()`, which waits (unchanged behavior) and always sets `grant` to `true`.

At the end of the first iteration we set `fTry`, so we either get a grant for the next iteration or we don't.

The first time we don't get a grant we bail out of the loop. This is effectively the same as before, it's just that we did `TryAquire()` at the start of each iteration (which also always would have succeed in the first iteration).